feat: translate README.md and navigation.md (#15)
elm-france/guide.elm-lang.org: book/webapps/url_parsing.md

# Parser les URLs

Dans la vraie vie, on veut que notre application montre des choses différentes selon les URLs :

- `/search`
- `/search?q=seiza`
- `/settings`

Oui, mais comment faire ? En utilisant [`elm/url`](https://package.elm-lang.org/packages/elm/url/latest/) pour parser ces chaînes brutes en structures de données en Elm. Le plus simple pour comprendre ce que fait ce paquet, c'est de regarder des exemples, alors faisons ça !

## Exemple n°1

> Disons que nous avons un site d'art, et que nous voulons proposer les adresses suivantes :

Reviewer: ```suggestion
Disons que nous avons un site d’art, et que nous voulons proposer les adresses suivantes :
```

- `/sujet/architecture`
- `/sujet/peinture`
- `/sujet/sculpture`
- `/blog/42`
- `/blog/123`
- `/blog/451`
- `/utilisateur/tom`
- `/utilisateur/sue`
- `/utilisateur/sue/commentaire/11`
- `/utilisateur/sue/commentaire/51`

> On a donc des pages sur des sujets, des posts de blog, des pages de profil, et un moyen d'afficher un commentaire d'utilisateur en particulier. On pourra utiliser le module [`Url.Parser`](https://package.elm-lang.org/packages/elm/url/latest/Url-Parser) pour écrire le parseur d'URL suivant :

Reviewer: ```suggestion
On a donc des pages sur des sujets, des posts de blog, des pages de profil, et un moyen d’afficher un commentaire d’utilisateur en particulier. On pourra utiliser le module [`Url.Parser`](https://package.elm-lang.org/packages/elm/url/latest/Url-Parser) pour écrire le parseur d’URL suivant :
```

-- /           ==>  Nothing
```

> Et hop, c'est bon pour les fragments !

Reviewer: ```suggestion
Et hop, c’est bon pour les fragments !
```

## Synthèse

> Maintenant qu'on a vu quelques parsers, regardongs comment tout ça s'intègre dans un programme `Browser.application`. Plutôt que de se contenter d'enregistrer l'URL comme la dernière fois, si on essayait de la transformer en donnée utile et de l'afficher ?

Reviewer: ```suggestion
Maintenant qu’on a vu quelques parsers, regardons comment tout ça s’intègre dans un programme `Browser.application`. Plutôt que de se contenter d’enregistrer l’URL comme la dernière fois, si on essayait de la transformer en donnée utile et de l'afficher ?
```

Reviewer: :point_up:

```elm
TODO
```

Les nouveautés :

> 1. notre fonction `update` parse l'URL quand elle reçoit un message `UrlChanged`.

Reviewer: ```suggestion
1. notre fonction `update` parse l’URL quand elle reçoit un message `UrlChanged` ;
```

2. notre fonction `view` montre des choses différentes selon les adresses !

Rien de bien compliqué. Tant mieux !

> Bon, mais il se passe quoi avec 10 pages différentes ? Ou 20 ? Ou 100 ? On met tout dans notre `view` ? On va pas laisser tout ça dans un seul fichier, si ? Mais alors, combien de fichiers ? Dans quelle arborescende de répertoires ? C'est ce qu'on va voir dans la partie suivante !

Reviewer: ```suggestion
Bon, mais il se passe quoi avec 10 pages différentes ? Ou 20 ? Ou 100 ? On met tout dans notre `view` ? On va pas laisser tout ça dans un seul fichier, si ? Mais alors, combien de fichiers ? Dans quelle arborescende de répertoires ? C’est ce qu'on va voir dans la partie suivante !
```
